### PR TITLE
Make evict leader scheduler compatitable 

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 
The result of the `scheduler` pdapi is in the following format as I showed below, the `store id` of the evict-leader-scheduler would be added at the end the of the scheduler name.

evict-2

However, in TiDB 4.0 release, the `store-id` would be omitted as I showed below.

evict-1

In this request, I add new method to make getting evict scheduler store id compatible between the current version and the coming 4.0 version.  

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: pkg/manager/member/pd_upgrader.go [hunk 1]
```diff
@@ -74,6 +74,7 @@ func (pu *pdUpgrader) gracefulUpgrade(tc *v1alpha1.TidbCluster, oldSet *apps.Sta
 
 	if controller.PodWebhookEnabled {
 		setUpgradePartition(newSet, 0)
+		return nil
```
Comment (Yisaer) [inline pkg/manager/member/pd_upgrader.go]: During my test, I found the bug in upgrading and I fixed it.

File: pkg/manager/member/tikv_upgrader.go [hunk 2]
```diff
@@ -89,6 +89,7 @@ func (tku *tikvUpgrader) Upgrade(tc *v1alpha1.TidbCluster, oldSet *apps.Stateful
 
 	if controller.PodWebhookEnabled {
 		setUpgradePartition(newSet, 0)
+		return nil
```
Comment (Yisaer) [inline pkg/manager/member/tikv_upgrader.go]: ditto.

File: pkg/webhook/pod/tikv_creater.go [hunk 3]
```diff
@@ -84,3 +89,37 @@ func (pc *PodAdmissionControl) admitCreateTiKVPod(pod *core.Pod, tc *v1alpha1.Ti
 
 	return util.ARSuccess()
 }
+
+// This is method is to make compatible between old pdapi version and 4.0 pdapi version.
```
Comment (DanielZhangQD) [inline pkg/webhook/pod/tikv_creater.go]:
```suggestion
// This method is to make compatible between old pdapi version and 4.0 pdapi version.
```

File: pkg/webhook/pod/tikv_creater.go [hunk 4]
```diff
@@ -84,3 +89,37 @@ func (pc *PodAdmissionControl) admitCreateTiKVPod(pod *core.Pod, tc *v1alpha1.Ti
...
+// To get more detail, see: https://github.com/pingcap/tidb-operator/pull/1831
+func filterLeaderEvictScheduler(evictLeaderSchedulers []string, pdClient pdapi.PDClient) (sets.String, error) {
+	schedulerIds := sets.String{}
+	if len(evictLeaderSchedulers) == 1 && evictLeaderSchedulers[0] == evictSchedulerLeader {
+		c, err := pdClient.GetConfig()
+		if err != nil {
+			return schedulerIds, err
+		}
+		if c.Schedule != nil {
+			if c.Schedule.SchedulersPayload == nil {
```
Comment (DanielZhangQD) [inline pkg/webhook/pod/tikv_creater.go]: If it's nil, why still need to continue to the following codes?

File: pkg/webhook/pod/tikv_creater.go [hunk 5]
```diff
@@ -84,3 +89,37 @@ func (pc *PodAdmissionControl) admitCreateTiKVPod(pod *core.Pod, tc *v1alpha1.Ti
...
+				c.Schedule.SchedulersPayload = map[string]string{}
+			}
+			v, ok := c.Schedule.SchedulersPayload[evictSchedulerLeader]
```
Comment (DanielZhangQD) [inline pkg/webhook/pod/tikv_creater.go]: Could you please give an example of the `c`? or any doc link about this new behavior?
I did not find the `c.Schedule.SchedulersPayload` in my cluster.

Comment (Yisaer) [reply pkg/webhook/pod/tikv_creater.go]: It's the new payload in pd 4.0

Comment (Yisaer) [reply pkg/webhook/pod/tikv_creater.go]: Here is an example for the payload response：
```bash
"balance-hot-region-scheduler": "null",
"balance-leader-scheduler": "{\"name\":\"balance-leader-scheduler\",\"ranges\":[{\"start-key\":\"\",\"end-key\":\"\"}]}",
"balance-region-scheduler": "{\"name\":\"balance-region-scheduler\",\"ranges\":[{\"start-key\":\"\",\"end-key\":\"\"}]}","evict-leader-scheduler": "{\"store-id-ranges\":{\"4\":[{\"start-key\":\"\",\"end-key\":\"\"}]}}",
"label-scheduler": "{\"name\":\"label-scheduler\",\"ranges\":[{\"start-key\":\"\",\"end-key\":\"\"}]}"

```

File: pkg/webhook/pod/tikv_creater.go (hunk 4 above)
Comment (Yisaer) [reply pkg/webhook/pod/tikv_creater.go]: Not necessary,  just solve the nil or empty payload with the same method.

File: pkg/webhook/pod/tikv_creater.go [hunk 6]
```diff
@@ -84,3 +89,37 @@ func (pc *PodAdmissionControl) admitCreateTiKVPod(pod *core.Pod, tc *v1alpha1.Ti
...
+	schedulerIds := sets.String{}
+	if len(evictLeaderSchedulers) == 1 && evictLeaderSchedulers[0] == evictSchedulerLeader {
+		c, err := pdClient.GetConfig()
+		if err != nil {
+			return schedulerIds, err
+		}
+		if c.Schedule != nil {
+			if c.Schedule.SchedulersPayload == nil {
+				c.Schedule.SchedulersPayload = map[string]string{}
+			}
```
Comment (DanielZhangQD) [inline pkg/webhook/pod/tikv_creater.go]: You can just change to:
```
if c.Schedule != nil && c.Schedule.SchedulersPayload != nil {
```

Comment (Yisaer) [reply pkg/webhook/pod/tikv_creater.go]: updatd, PTAL

**Pull request merged**